File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/domain/common/Result.java
package com.crpr.androidcinema.domain.common;

/**
 * Created by cribeiro on 14/01/2016.
 */
public class Result {

    public static final int OK = 1;
    public static final int ERROR = 2;
    private final int _resultStatusCode;
    private String _message;

    public Result(int statusCode){
        _resultStatusCode = statusCode;
        _message = "";
    }

    public Result(int statusCode, String message){
        _resultStatusCode = statusCode;
        _message = message;
    }

    public String getMessage(){
        return _message == null ? "" : _message;
    }

    public boolean hasError(){
        return this._resultStatusCode == ERROR;
    }
}

File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/domain/common/Result.kt
package com.crpr.androidcinema.domain.common

/**
 * Created by cribeiro on 14/01/2016.
 */
open class Result {
    private val _resultStatusCode: Int
    private var _message: String? = null

    constructor(statusCode: Int) {
        _resultStatusCode = statusCode
        _message = ""
    }

    constructor(statusCode: Int, message: String) {
        _resultStatusCode = statusCode
        _message = message
    }

    val message: String
        get() = if (_message == null) "" else _message!!

    fun hasError(): Boolean {
        return this._resultStatusCode == ERROR
    }

    companion object {

        val OK = 1
        val ERROR = 2
    }
}
